src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -31,6 +32,12 @@ public StartableEndpoint(SettingsHolder settings, IBuilder builder, FeatureActiv
 
         public async Task<IEndpointInstance> Start()
         {
+            var result = await receiveInfrastructure.PreStartupCheck().ConfigureAwait(false);
```

Comment:
While this PR is focused on the receive feature, I'd like to see the same thing happen for sending as well because I'd like to see both startup checks here.